Zend/zend_vm_def.h

```diff
@@ -1487,6 +1487,13 @@ ZEND_VM_HELPER(zend_pre_inc_helper, VAR|CV, ANY)
 	SAVE_OPLINE();
 	if (OP1_TYPE == IS_CV && UNEXPECTED(Z_TYPE_P(var_ptr) == IS_UNDEF)) {
 		ZVAL_UNDEFINED_OP1();
+		if (UNEXPECTED(EG(exception))) {
+			/* Smart branch expects result to be set with exceptions */
```

Comment:
I didn't understand this comment.
PRE/POST_INC/DEC are never combined with the following JMPZ/NZ into smart branches.

Reply:
It's actually about this line:

https://github.com/php/php-src/blob/66b359e4de3e0f4f49b8a213407f0bf4b9a7fb11/Zend/zend_vm_def.h#L8099

The problem is not related to smart branches. Instead, it's every op but smart branches, that are expected to initialize the result (apart from some others that don't store zvals). If the result is not initialized, we'll `free` uninitialized value. With opcache, we can get a use-after-free if the zval still contains some old value due to shared VARs.